spyder/api/plugins.py

```diff
@@ -748,6 +748,18 @@ class SpyderPluginV2(QObject, SpyderActionMixin, SpyderConfigurationObserver):
     This signal can be emitted to request the main application to restart.
     """
 
+    sig_restart_reset_requested = Signal()
+    """
+    This signal can be emitted to request the main application to restart with
+    reset preferences.
+    """
+
+    sig_restart_debug_requested = Signal()
+    """
+    This signal can be emitted to request the main application to restart in
+    DEBUG mode.
+    """
+
```

Comment:
If we want to move all the restart logic to the Application plugin, I think having these signals (and `sig_restart_requested`) makes little sense. Instead, plugins that need to perform a restart should require Application and call their restart methods directly.

However, if we want to keep using signals, we should preserve the current mechanism, i.e. connecting slots in the main window to those signals as we do now:

https://github.com/spyder-ide/spyder/blob/77d11c006dd197a05bb029c9310b33c4c222c4b3/spyder/app/mainwindow.py#L224

Personally, I find that logic very convoluted and hard to follow: plugins need to emit signals, those signals are connected in the main window to slots that end up calling Application methods. So why not call those methods directly?

The current problem with that is that the Preferences plugin would require Application to perform a restart,  but Application requires Preferences to register its config page. So that would introduce a circular dependency between those plugins.

The good thing is that problem will be avoided when we merge PR https://github.com/spyder-ide/spyder/pull/15582 because it will remove our current linear dependency plugin mechanism for a notification based one.

@mrclary, as someone who has tried to understand how this code works, what would you prefer? I mean, using signals or calling Application methods directly?

Reply:
@ccordoba12, perhaps we can schedule a zoom discussion?

Reply:
@ccordoba12, I don't feel strongly one way or the other. From your previous suggestion to move actions to the plugin, I (apparently incorrectly) thought that included the signal handling. I can move things back as before, keeping the signal handling in the main window, but putting actions in the plugin.

Reply:
> @mrclary will try to give it a check 👍

@dalthviz, thanks, but hold off since I'll be making changes again.

Reply:
@mrclary, we'll discuss this topic with the rest of the team and have an answer for you soon. Please don't make more changes until then.

Reply:
@ccordoba12, I'd like to be part of that discussion; will we schedule a zoom meeting or something?

Reply:
Good thinking! Sorry, I didn't want to make feel excluded, I just didn't know you wanted to be part of that discussion.

Could you make it tomorrow at 12 pm central time? Or perhaps this Monday at 13:00, central time too?

Reply:
I can do both, thanks!

Reply:
Great! I'll send you the invite for Monday then because it's easier for me.